src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts

```diff
@@ -180,6 +192,28 @@ function prepareNavNodeForClient({
 			// Otherwise, this is a genuinely external NavDirectLink,
 			// so we only need to add an `id` to it.
 			const preparedItem = { ...node, id }
+
+			/**
+			 * Rewrite external Learn and Docs links if needed.
+			 * 	- learn.hashicorp.com
+			 * 	- vaultproject.io
+			 * 	- waypointproject.io
+			 */
+			try {
+				const urlObject = new URL(node.href)
+				if (getIsExternalLearnLink(node.href)) {
+					preparedItem.href = rewriteExternalLearnLink(urlObject, TUTORIAL_MAP)
```

Comment:
Might be good to add a protection here to make sure `TUTORIAL_MAP` is defined, similar to above.

Reply:
Oh right! Just in case 😊

Reply:
Updated in https://github.com/hashicorp/dev-portal/pull/1052/commits/d003c2ea8aa41614f56828b19f068558c75a138c! ✅

Reply:
Sorry I meant the actual `TUTORIAL_MAP`. Should we check that it is defined and it not, generate it if there is a  possibility of hitting this conditional first?

Reply:
Oh! I don't think that can happen since only `prepareNavDataForClient` calls this function and it loads the `TUTORIAL_MAP` before doing so.

Reply:
okay perfect 😸